src/Microsoft.ML.Core/Data/ReadOnlyMemoryUtils.cs

```diff
@@ -0,0 +1,325 @@
+        {
+            if (memory.IsEmpty)
+            {
+                left = memory;
+                right = default;
+                return false;
+            }
+
+            int index = memory.Span.IndexOf(separator);
+            if (index == -1)
+            {
+                left = memory;
+                right = default;
+                return false;
+            }
+
+            left = memory.Slice(0, index);
+            right = memory.Slice(index + 1, memory.Length - index - 1);
+            return true;
+        }
+
+        /// <summary>
+        /// Splits <paramref name="memory"/> on the left-most occurrence of an element of separators character array and
+        /// produces the left and right <see cref="ReadOnlyMemory{T}"/> of <see cref="char"/> values. If <paramref name="memory"/> does not contain any of the
+        /// characters in separators, this return false and initializes <paramref name="left"/> to this instance
+        /// and <paramref name="right"/> to the default <see cref="ReadOnlyMemory{T}"/> of <see cref="char"/> value.
+        /// </summary>
+        public static bool SplitOne(ReadOnlyMemory<char> memory, char[] separators, out ReadOnlyMemory<char> left, out ReadOnlyMemory<char> right)
+        {
+            Contracts.CheckValueOrNull(separators);
+
+            if (memory.IsEmpty || separators == null || separators.Length == 0)
+            {
+                left = memory;
+                right = default;
+                return false;
+            }
+
+            int index;
+            if (separators.Length == 1)
+                index = memory.Span.IndexOf(separators[0]);
+            else
+                index = memory.Span.IndexOfAny(separators);
+
+            if (index == -1)
+            {
+                left = memory;
+                right = default;
+                return false;
+            }
+
+            left = memory.Slice(0, index);
+            right = memory.Slice(index + 1, memory.Length - index - 1);
+            return true;
+        }
+
+        /// <summary>
+        /// Returns a <see cref="ReadOnlyMemory{T}"/> of <see cref="char"/> with leading and trailing spaces trimmed. Note that this
+        /// will remove only spaces, not any form of whitespace.
+        /// </summary>
+        public static ReadOnlyMemory<char> TrimSpaces(ReadOnlyMemory<char> memory)
+        {
+            if (memory.IsEmpty)
+                return memory;
+
+            int ichLim = memory.Length;
+            int ichMin = 0;
+            var span = memory.Span;
+            if (span[ichMin] != ' ' && span[ichLim - 1] != ' ')
+                return memory;
+
+            while (ichMin < ichLim && span[ichMin] == ' ')
+                ichMin++;
+            while (ichMin < ichLim && span[ichLim - 1] == ' ')
+                ichLim--;
+            return memory.Slice(ichMin, ichLim - ichMin);
+        }
+
+        public static ReadOnlySpan<char> TrimSpaces(ReadOnlySpan<char> span)
```

Comment:
Use the existing Trim extension method overload that accepts a trim character.

```C#
span.Trim(' ');
```

The same would apply to the ReadOnlyMemory utility method once we add the new Trim method.
https://github.com/dotnet/corefx/issues/30592